cpp/include/ray/api/ray_config.h

```diff
@@ -48,6 +48,9 @@ class RayConfig {
 
   // Prevents external clients without the password from connecting to Redis if provided.
   boost::optional<std::string> redis_password_;
+
+  // The default actor lifetime type, `detached` or `non_detached`.
+  std::string default_actor_lifetime = "non_detached";
```

Comment:
If this is a stable config, move it to line 48. Otherwise, rename it from `default_actor_lifetime` to `default_actor_lifetime _`.

Reply:
default_actor_lifetime -> job_default_actor_lifetime?

Reply:
1. I will move it to line 48.
2. default_actor_lifetime is public config. I think using default_actor_lifetime is better.
3. Both java and python are named this way. Keep the same name.

Reply:
We use `default_actor_lifetime` in java because it is in `JobConfig`. In C++ currently, we don't have `JobConfig` class. So, I suggest use a prefix `job_` to distinguish it. I'm not sure if it is the best way. But we should address. @raulchen What do you think?

Reply:
Now only python have jobconfig. Java also is RayConfig.

Reply:
@larrylian That Java "RayConfig" class is not a user-facing API. So the naming of those fields doesn't matter. The actual Java user API for this feature is `System.setProperty("ray.job.default-actor-lifetime", "DETACHED")`, which does have the word "job" in it. 
I think @SongGuyang's argument makes sense. But I think it's better to introduce a "JobConfig" and put all job-related items in it.

Reply:
Add an issue here https://github.com/ray-project/ray/issues/25989. Don't block current PR.